### DISCUSSION TITLE
Response validation fails due to missing optional elements

### DISCUSSION THREAD

**Discussion by DerManoMann** [luolingchun/flask-openapi3#50] [Q&A]
Hi,

I do have a data type using `Optional` like in the example below. What is wrong here?

```python
class PackageData(BaseModel):
    """Single package data."""

    id: UUID = Field(..., description="UUID")
    title: str = Field(..., description="Title")
    visibility: Optional[str] = Field(..., description="Visibility of the package")
```

Tests are failing with this, though:
```
E   visibility
E     field required (type=value_error.missing)
```

**Comment by luolingchun** [comment]
... means required. You can use None or other strings.

**Comment by DerManoMann** [comment]
I was under the impression the `Optional[str]` for `visibility` would flag it as optional property...

So, is there no way to have optional properties in the schema? I am sure I've seen that using other languages...

**Comment by luolingchun** [comment]
I guess you need official documents: https://pydantic-docs.helpmanual.io/usage/models/#required-optional-fields
